docs/concept-workflows.md

```diff
@@ -1,30 +1,209 @@
+
+But, by using event sourcing a Workflow becomes quite durable and in essence "fault-tolerant".
+For example, if a host fails in the middle of a Workflow execution, not only can another host resume the Workflow, but it can resume at the exact line of code where it stopped executing without having to re-execute the code up until that point.
+The Workflow execution state is even preserved through crashes of the Temporal Server itself.
+
+## Freedom to relocate
+
+While there are a few specific use-cases that may require a Workflow to start and finish on the same host, Workflow code is completely independent from the location in which it is executed
+
+Instead of deploying executable code to a host directly, Temporal applications use special [Worker](/docs/concept-workers) processes whose sole job is to host and execute Workflow and Activity code.
+The Worker communicates with the Temporal Server in order to know which line of code to execute through [Tasks](/docs/concept-task-queues) it picks up.
+
+Because a Workflow is executed piece by piece, and because the Server tracks the state of that execution, there can be multiple Worker processes in multiple locations capable of executing any one of those pieces.
+
+Essentially load balancing and processing failover are "out-of-the-box" features of Temporal Workflows.
+
+## Workflows have options
+
+When you start a Workflow, you can pass along parameters that tell the Temporal Server how to handle the Workflow.
+This includes the ability to set timeouts for Workflow execution, a retry policy, the Task Queue name, a data converter, and child Workflow options.
+
+### Timeout settings
+
+It's often necessary to limit the amount of time a specific Workflow can be running.
+
+To support this, the following three parameters can be provided to Workflow options:
+
+#### Execution timeout
+
+- **Description**: This is the maximum amount of time that a Workflow should be allowed to run including retries and any usage of the "Continue-as-new" feature.
+This is different from [Run timeout](#run-timeout).
+- **Use-case**:
+
+#### Run timeout
+
+- **Description**: This is the maximum amount of time that a single Workflow run is restricted to.
+- **Use-case**:
+
+#### Task timeout
+
+- **Description**: This is the maximum amount of time that the Server will wait for the Worker to start processing a Task after the Task has been pulled from the Task Queue.
+If a [Command Task](/docs/glossary/#command) is lost, it is retried after this timeout.
+- **Use-case**:
+
+### Retry policy
+
+There may be scenarios where you need to retry a Workflow's execution from the very beginning.
+In this case, you can use supply a retry policy when you start the Workflow.
+
+#### Initial interval
+
+- **Description**: Amount of time that must elapse before the first retry occurs.
+- **Use-case**:
```

Comment:
@mfateev ?

Reply:
As a general rule of thumb, we recommend writing workflows to not fail on intermittent issues. So retrying workflows is practically never needed. There are some exceptions like cron workflows or some other mostly stateless always running workflows that can benefit from workflow retries.